staging/src/k8s.io/cli-runtime/pkg/printers/yaml.go

```diff
@@ -28,6 +28,9 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// globalPrintCount records the total count to support when printer is called multi-times separately.
+var globalPrintCount int64
```

Comment:
There is an earlier similar attempt about this change https://github.com/kubernetes/kubernetes/pull/112455. I think we need to find a way without introducing a new global flag(even not touching this package). Because all commands are using this and I don't see any command based counter management in the code(what happens when I run multiple `kubectl label` commands and subsequent one `kubectl annotate` command, does `kubectl annotate` command contains separator. Because it should not?).

Reply:
Agreed with Arda--we cannot introduce a global variable for this.

Reply:
Got it, updated with no change in yaml printer.
Then this PR will not fix https://github.com/kubernetes/kubectl/issues/1323.

Reply:
Actually this PR will not fix this one https://github.com/kubernetes/kubectl/issues/1265

Reply:
Yes, typo updated.

Reply:
I think we do need to fix that one, in the sense that if we simply copy over the existing implementation over to annotate, we will be bringing that bug along with it. I don't think we should knowingly do that. And once we have the solution, we might as well fix it in label at the same time.

Reply:
OK, will try to fix it.